src/features/swap/index.tsx

```diff
@@ -61,62 +61,52 @@ const SwapWidget = ({ sell }: Params) => {
   const wallet = useWallet()
   const { isConsentAccepted, onAccept } = useSwapConsent()
 
-  const groupKey = 'swap-order-status'
   const listeners = useMemo<CowEventListeners>(() => {
     return [
       {
         event: CowEvents.ON_TOAST_MESSAGE,
         handler: (event) => {
           console.info('🍞 New toast message:', event)
           const { messageType } = event
+
           switch (messageType) {
             case 'ORDER_CREATED':
               dispatch(
-                showNotification({
-                  title: 'Swap transaction created',
-                  message: 'Waiting for confirmation from signers of your Safe',
-                  groupKey,
-                  variant: 'info',
+                setSwapOrder({
+                  orderUid: event.data.orderUid,
+                  status: 'created',
```

Comment:
I don't think we need a switch case here since we always call the same function. Instead we could pass the `messageType`:

```
dispatch(
    setSwapOrder({
        orderUid: event.data.orderUid,
        status: messageType,
    })
)
```

Or if we want to have a different status notation we could add a mapper to go from `ORDER_CREATED -> created` etc.

Reply:
Not that easy. You would at least need a mapping of the cow eventListener type to the allStatusesType (which is the cgw status type + few more)